11/28/2025 - Nafisa - Load all raw data files
**Purpose:**  
Load all CSV, JSON, and Parquet files from the fall2025_L directory, merge them into a single DataFrame, clean labels, and save the combined dataset for downstream processing.


**Interpretation / Findings:** 
- Total 5 unique labels found: BENIGN, DoS Hulk, DoS GoldenEye, DoS Slowhttptest, Heartbleed.
- Heartbleed (11 rows) removed as required.
- Created binary target column Attack → 1 for attacks, 0 for benign.
- Final dataset size after cleaning:
    -  Attack = 1: 56,112 rows
    - Attack = 0: 5,005 rows
- Output saved to: data/interm/combined_raw.csv

**Notes for Team:**
The combined dataset is now ready for feature engineering and model training. Please ensure to review the cleaned labels and the binary target column for consistency before proceeding. 

### Done with Task

In [1]:
import pandas as pd

# read in the dataset 
df = pd.read_csv('../data/interm/combined_raw.csv')

# Count each label amount 
label_counts = df[' Label'].value_counts()
print("Label Counts:")
print(label_counts)
print()

# Count the total amount of DOS attacks 
total_dos = label_counts[label_counts.index != 'BENIGN'].sum()
print(f"Total DOS attacks: {total_dos}")
print(f"Total BENIGN: {label_counts['BENIGN']}")
print(f"Total records: {len(df)}")

Label Counts:
 Label
DoS Hulk            30027
DoS GoldenEye       20586
DoS Slowhttptest     5499
BENIGN               5005
Name: count, dtype: int64

Total DOS attacks: 56112
Total BENIGN: 5005
Total records: 61117


### Label Distribution

| Label | Count |
|-------|-------|
| DoS Hulk | 30,027 |
| DoS GoldenEye | 20,586 |
| DoS Slowhttptest | 5,499 |
| BENIGN | 5,005 |

**Summary:**
- Total DOS attacks: 56,112
- Total BENIGN: 5,005
- Total records: 61,117

DONE WITH TASK 